compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -442,6 +442,9 @@ object Types {
     final def containsWildcardTypes(using Context) =
       existsPart(_.isInstanceOf[WildcardType], StopAt.Static, forceLazy = false)
 
+    /** Is this a union type? */
+    final def isOrType: Boolean = this.isInstanceOf[OrType]
```

Comment:
It's a bit dubious to add a local test like that. What about AndType? What about all other occurrences of `.isInstanceOf[OrType]`? The most important goal to stay sane is consistency!